### ISSUE TITLE
Documentation how to use ToggleMonitor functionality

### ISSUE THREAD

**Issue by kpbochenek** [travitch/taffybar#215]
If it doesn't fit to 'issues' please let me know where I could ask about this.

I noticed that it was added something like "multi monitor support" yet I cannot get to understand how this works.

It would be awesome to add section to documentation how it can be used.

E.g. in my case I have laptop where by default I want to have taffybar running with only single monitor present.
But sometimes I plug in external monitor and I want taffybar to move to that external monitor.
To enable external monitor I run my script and I can add additional instructions there but still have no idea what would that be.
Replaced 'defaultTaffybar' with 'withToggleSupport' but that would be it :(

What 'toggleState.hs' is for and how example file should look like. 

Also it would be great to provide example regarding dbus.

Thanks!
Krzysiek

**Comment by IvanMalison** [comment]
Hi @kpbochenek, I'm the author of this feature. I agree that some documentation would be a good idea. I'll try to add some in the future, but for now I can explain how I use this here:

This command will toggle taffybar on whatever monitor is currently 'active':
```
dbus-send --print-reply=literal --dest=taffybar.toggle /taffybar/toggle taffybar.toggle.toggleCurrent
```

You can use a command like

```
dbus-send --print-reply=literal --dest=taffybar.toggle /taffybar/toggle hidebar.toggle.showOnMonitor int32:1
```
To explicitly enable taffybar on a monitor (but you'll need to know the number it has been assigned)

toggleState.hs is a file that is used to restore the previous state of taffybar when it is restarted. You should not edit it unless you are very sure of what you are doing.

**Comment by kpbochenek** [comment]
Getting closer to make this work! :)

Got exception when tried to toggle current monitor(in a third or fourth time so partially works):
```
[xcb] Unknown sequence number while processing queue
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
taffybar-linux-x86_64: xcb_io.c:259: poll_for_event: Assertion`!xcb_xlib_threads_sequence_lost' failed.
[1]    6032 abort (core dumped)  stack exec taffybar
```

Any idea why?

**Comment by kpbochenek** [comment]
In all cases it is when toggleCurrent has to hide taffybar on screen.

**Comment by IvanMalison** [comment]
> Got exception when tried to toggle current monitor(in a third or fourth time so partially works):

Yeah I've been dealing with this myself recently as well.



> Any idea why?

My guess was originally that it had to do with things not being properly on the main gtk thread see
https://github.com/gtk2hs/gtk2hs/blob/21d3a12a82756d04066e5e4198b35097012ee65b/gtk/Graphics/UI/Gtk/General/General.chs#L136

, but I did add a postGuiSync to the refresh action, so it seems like that should not be the issue:
https://github.com/travitch/taffybar/blob/3a5024368b4783084302b61cb6eb0ba27e305693/src/System/Taffybar.hs#L428

I'm taking a peek at this again, because it is starting to annoy me.

Just to make sure, you are able to accomplish the desired effect sometimes, but you also get crashes sometimes, right?

**Comment by IvanMalison** [comment]
Great news @kpbochenek! I figured out a fix for this issue in #216

**Comment by kpbochenek** [comment]
@IvanMalison looks stable now!

Now time to peek into code what dbus commands are available :)
